src/main/scala/firrtl/traversals/Foreach.scala

```diff
@@ -0,0 +1,108 @@
+package firrtl.traversals
+
+import firrtl.ir._
+
+object Foreachers {
+
+  // ********** Stmt Mappers **********
+  private trait StmtForMagnet {
+    def foreach(stmt: Statement): Unit
+  }
+  private object StmtForMagnet {
+    implicit def forStmt(f: Statement => Unit): StmtForMagnet = new StmtForMagnet {
+      override def foreach(stmt: Statement): Unit = stmt foreachStmt f
```

Comment:
Also copy-pasta, but is there any reason why this does an override? Is this trying to conform to the stricter Databricks-like style of "use override if you're overriding to write more readable code"? 

I'm fine with it either way.